hoss/coordinate_utilities.py

```diff
@@ -0,0 +1,232 @@
+) -> Tuple[int, int]:
+    """
+    This method returns the row and column sizes of the coordinate datasets
+
+    """
+    if lat_arr.ndim > 1 and lon_arr.shape == lat_arr.shape:
+        col_size = lat_arr.shape[1]
+        row_size = lat_arr.shape[0]
+    elif (
+        lat_arr.ndim == 1
+        and lon_arr.ndim == 1
+        and lat_arr.size > 0
+        and lon_arr.size > 0
+    ):
+        col_size = lon_arr.size
+        row_size = lat_arr.size
+    else:
+        raise IrregularCoordinateVariables(lon_arr.shape, lat_arr.shape)
+    return row_size, col_size
+
+
+def get_lat_lon_arrays(
+    prefetch_dataset: Dataset,
+    latitude_coordinate: VariableFromDmr,
+    longitude_coordinate: VariableFromDmr,
+) -> Tuple[ndarray, ndarray]:
+    """
+    This method is used to return the lat lon arrays from a 2D
+    coordinate dataset.
+    """
+    try:
+        lat_arr = prefetch_dataset[latitude_coordinate.full_name_path][:]
+    except Exception as exception:
+        raise MissingCoordinateVariable(
+            latitude_coordinate.full_name_path
+        ) from exception
+
+    try:
+        lon_arr = prefetch_dataset[longitude_coordinate.full_name_path][:]
+    except Exception as exception:
+        raise MissingCoordinateVariable(
+            longitude_coordinate.full_name_path
+        ) from exception
+
+    return lat_arr, lon_arr
+
+
+def get_dimension_scale_from_dimvalues(
+    dim_values: ndarray, dim_indices: ndarray, dim_size: float
+) -> ndarray:
+    """
+    return a full dimension scale based on the 2 projected points and
+    grid size
+    """
+    dim_resolution = 0.0
+    if (dim_indices[1] != dim_indices[0]) and (dim_values[1] != dim_values[0]):
+        dim_resolution = (dim_values[1] - dim_values[0]) / (
+            dim_indices[1] - dim_indices[0]
+        )
+    if dim_resolution == 0.0:
+        raise CannotComputeDimensionResolution(dim_values[0], dim_indices[0])
+
+    # create the dim scale
+    dim_asc = dim_values[1] > dim_values[0]
+
+    if dim_asc:
+        dim_min = dim_values[0] + (dim_resolution * dim_indices[0])
+        dim_max = dim_values[0] + (dim_resolution * (dim_size - dim_indices[0] - 1))
+        dim_data = np.linspace(dim_min, dim_max, dim_size)
+    else:
+        dim_max = dim_values[0] + (-dim_resolution * dim_indices[0])
+        dim_min = dim_values[0] - (-dim_resolution * (dim_size - dim_indices[0] - 1))
+        dim_data = np.linspace(dim_max, dim_min, dim_size)
+
+    return dim_data
+
+
+def get_valid_indices(
+    coordinate_row_col: ndarray, coordinate_fill: float, coordinate_name: str
+) -> ndarray:
+    """
+    Returns indices of a valid array without fill values
+    """
+
+    if coordinate_fill:
+        valid_indices = np.where(
+            ~np.isclose(coordinate_row_col, float(coordinate_fill))
+        )[0]
+    elif coordinate_name == 'longitude':
+        valid_indices = np.where(
+            (coordinate_row_col >= -180.0) & (coordinate_row_col <= 180.0)
```

Comment:
oh sorry I knew that. Not sure why I used &. will fix.

Reply:
I think changing to `and` gave a different error : 
Simplify chained comparison between the operands (chained-comparison)

have to change that check to be simpler

Reply:
I just wanted to come back to this thread - I was wrong about the `&`. But I think I prefer the equivalent `np.logical_and` just because I'm generally too verbose.